src/proto/streams/send.rs

```diff
@@ -380,15 +380,24 @@ impl Send {
 
                     stream.send_flow.dec_window(dec);
 
+                    let window_size = stream.send_flow.window_size();
                     let available = stream.send_flow.available().as_size();
-                    let reclaim = cmp::min(dec, available);
-                    stream.send_flow.claim_capacity(reclaim);
-                    total_reclaimed += reclaim;
+
+                    let reclaimed = if available > window_size {
```

Comment:
I admittedly no longer remember exactly what should be done. I wish we had let more comments relating back to the spec about what is done here. I know you're just fixing broken things, but since you're in here and seem to have more context, would it be possible to add a few lines of comments reminding Future Us what is supposed to happening? If not, not a blocker at all!